src/beamlime/applications/interfaces.py

```diff
@@ -228,185 +80,39 @@ def output_channel(self):
     def output_channel(self, output_channel):
         self._output_ch = output_channel
 
-    @async_timeout(Empty)
-    async def receive_data(self, *args, timeout=10, wait_interval=0.2, **kwargs):
-        # TODO: Move async_timeout(exception=Empty) to communication handler interface
-        # and remove the decorator or use @async_timeout(exception=TimeoutError).
-        return self.input_channel.get(*args, timeout=wait_interval, **kwargs)
+    async def receive_data(self, *args, **kwargs) -> Any:
```

Comment:
This interface has `receive_data` and `send_data` methods defined, which are specified in the `BeamlimeDownstreamProtocol`. Should it be both a `BeamlimeApplicationProtocol` and a `BeamlimeDownstreamProtocol`?

Reply:
Yes...! They will be done with communication interfaces...!